techlibs/ice40/ice40_opt.cc

```diff
@@ -141,6 +141,9 @@ static void run_ice40_opts(Module *module)
 				cell->type = "$lut";
 				cell->setPort("\\A", { cell->getPort("\\I0"), inbit[0], inbit[1], cell->getPort("\\I3") });
 				cell->setPort("\\Y", cell->getPort("\\O"));
+				auto LUT = cell->getParam(ID(LUT_INIT));
+				cell->setParam("\\LUT", std::vector<State>{LUT[0], LUT[8], LUT[4], LUT[12], LUT[2], LUT[10], LUT[6], LUT[14], LUT[1], LUT[9], LUT[5], LUT[13], LUT[3], LUT[11], LUT[7], LUT[15]});
```

Comment:
I think this will crash if LUT_INIT is less than 16 bits to start with - something like `cell->getParam(ID(LUT_INIT)).extract(0, 16)` would probably be better here

Reply:
Good spot! Is the behaviour of an under-specified `LUT_INIT` written down anywhere? `Const::extract()` would (by default at least) pad these bits with zero -- is this the right behaviour?

Reply:
Okay I've set the padding to be 1'bx in ccd8f9e6f